src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeLoaderEnvironment.Metadata.cs

```diff
@@ -823,11 +823,38 @@ public struct VirtualResolveDataResult
             ExternalReferencesTable externalReferences = default(ExternalReferencesTable);
             externalReferences.InitializeCommonFixupsTable(module);
 
```

Comment:
Fixed for the declaring type semantics. No changes done to the slot index computation as we discussed, since this code is only used with SUPPORTS_NATIVE_METADATA_TYPE_LOADING